src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ValueListBuilder.cs

```diff
@@ -100,16 +100,29 @@ public void Insert(int index, scoped ReadOnlySpan<T> source)
             _pos += source.Length;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Span<T> AppendSpan(int length)
         {
             Debug.Assert(length >= 0);
 
             int pos = _pos;
-            if ((uint)(pos + length) > (uint)_span.Length)
+            Span<T> span = _span;
+            if ((ulong)(uint)pos + (ulong)(uint)length <= (ulong)(uint)span.Length) // same guard condition as in Span<T>.Slice on 64-bit
```

Comment:
```suggestion
#if TARGET_64BIT
            // See comment in Span<T>.Slice for how this works.
            if ((ulong)(uint)pos + (ulong)(uint)length <= (ulong)(uint)span.Length)
#else
            if ((uint)pos <= (uint)span.Length && (uint)length <= (uint)(span.Length - pos))
#endif
```

Reply:
ValueListBuilder is used in other assemblies as well.